frontend/public/components/cloud-services/catalog-source.tsx

```diff
@@ -3,16 +3,34 @@
 import * as React from 'react';
 import * as _ from 'lodash-es';
 import { match, Link } from 'react-router-dom';
-import { Helmet } from 'react-helmet';
 import { safeLoad } from 'js-yaml';
 
-import { NavTitle, SectionHeading, Firehose, MsgBox, LoadingBox, withFallback } from '../utils';
+import { SectionHeading, Firehose, MsgBox, LoadingBox, withFallback, ResourceCog, ResourceLink, Cog, navFactory, resourceObjPath } from '../utils';
+import { fromNow } from '../utils/datetime';
 import { CreateYAML } from '../create-yaml';
-import { ClusterServiceVersionLogo, SubscriptionKind, CatalogSourceKind, ClusterServiceVersionKind, Package } from './index';
-import { SubscriptionModel, CatalogSourceModel } from '../../models';
-import { referenceForModel, K8sResourceKind, apiVersionForModel } from '../../module/k8s';
-import { List, ListHeader, ColHead } from '../factory';
+import { ClusterServiceVersionLogo, CatalogSourceKind, ClusterServiceVersionKind, Package, SubscriptionKind, olmNamespace } from './index';
+import { SubscriptionModel, CatalogSourceModel, ConfigMapModel } from '../../models';
+import { referenceForModel, K8sResourceKind, apiVersionForModel, ConfigMapKind } from '../../module/k8s';
+import { List, ListHeader, ColHead, ResourceRow, DetailsPage, MultiListPage } from '../factory';
 import { registerTemplate } from '../../yaml-templates';
+import { getActiveNamespace } from '../../ui/ui-actions';
+import { ALL_NAMESPACES_KEY } from '../../const';
+
+type ConfigMapFor = (data: {data?: ConfigMapKind[]}) => (name: string) => ConfigMapKind;
+const configMapFor: ConfigMapFor = data => name => _.get(data, 'data', [] as ConfigMapKind[]).find(cm => cm.metadata.name === name);
+
+type SubscriptionsFor = (subs: SubscriptionKind[]) => (obj: CatalogSourceKind) => SubscriptionKind[];
+const subscriptionsFor: SubscriptionsFor = subs => obj => subs.filter(sub => sub.spec.source === obj.metadata.name);
+
+type PackagesFor = (configMap: ConfigMapKind) => (obj: CatalogSourceKind) => Package[];
+const packagesFor: PackagesFor = configMap => obj => _.get(configMap, 'data.packages')
+  ? safeLoad(_.get(configMap, 'data.packages'))
+  : [];
+
+type ClusterServiceVersionsFor = (configMap: ConfigMapKind) => (obj: CatalogSourceKind) => ClusterServiceVersionKind[];
+const clusterServiceVersionsFor: ClusterServiceVersionsFor = configMap => obj => _.get(configMap, 'data.clusterServiceVersions')
+  ? safeLoad(_.get(configMap, 'data.clusterServiceVersions'))
```

Comment:
Do we need to catch exceptions in these calls to safe load?

Reply:
True, I've wrapped the components that use `safeLoad` with React error boundaries using `withFallback` HOC.